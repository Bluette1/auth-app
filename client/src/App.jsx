import { Routes, Route } from 'react-router-dom';
import { useSelector } from 'react-redux';
import Home from './components/Home';
import Profile from './components/Profile';
import './styles/App.css';

function App() {
  const { loggedIn } = useSelector((state) => state.auth);

  return (
    <div data-testid="app">
      <Routes>
        <Route element={loggedIn ? <Profile /> : <Home />} path="/" />
        <Route path="/profile" element={loggedIn ? <Profile /> : <Home />} />
      </Routes>
    </div>
  );
}

export default App;
