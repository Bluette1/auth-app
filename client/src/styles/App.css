@import url('https://fonts.googleapis.com/css2?family=Open+Sans&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Open+Sans&family=Poppins:wght@400;600&display=swap');
html {
  box-sizing: border-box;
}

*,
*::before,
*::after {
  box-sizing: inherit;
}

* {
  border: 0 none;
  list-style: none outside none;
  margin: 0;
  outline: 0 none;
  padding: 0;
}

html {
  height: 100%;
}

body {
  max-width: 100%;
  margin: 0 auto;
  font-size: 1rem;
  line-height: 1.25rem;
  background-image: url('../assets/login_background_desktop.png');
  background-repeat: no-repeat;
  background-color: rgb(203, 203, 227);
  font-family: 'Open Sans', sans-serif;
  height: fit-content;
}

a {
  text-decoration: none;
  color: black;
}

a:hover {
  text-decoration: none;
  color: inherit;
}

.logo-area {
  display: flex;
  justify-content: center;
}

img.logo {
  width: 15rem;
  height: 18.5rem;
}

main {
  margin: 2%;
  display: flex;
  justify-content: center;
}

main > article {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.icons {
  height: 40px;
}

h1 {
  font-family: Poppins;
  font-style: normal;
  font-weight: 600;
  font-size: 2.5rem;
  line-height: 3.75rem;
  color: #ffffff;
}

button {
  width: 12.5rem;
  height: 3.5rem;
  font-family: Poppins;
  font-style: normal;
  font-weight: normal;
  font-size: 1.125rem;
  line-height: 1.688rem;
  text-align: center;
  border-radius: 14.5rem;
  margin-top: 7.5rem;
  background-color: #ffffff;
  cursor: pointer;
}

button:hover {
  color: #ffffff;
  background-color: rgb(164, 147, 114);
}

.attribution {
  font-size: 11px;
  text-align: center;
  margin-top: 14.75%;
}

.attribution a {
  color: #f7f3f5;
}

img {
  max-width: 100%;
  min-width: 300px;
  height: auto;
}

.absolute-bg {
  background-size: cover;
  background-image: linear-gradient(
      to bottom,
      rgba(32, 136, 136, 0.4),
      rgba(193, 205, 193, 0.8),
      rgb(164, 147, 114)
    ),
    url('../assets/login_background_desktop.png');
}

@media only screen and (max-width: 768px) {
  body {
    background-image: url('../assets/login_background_mobile.png');
  }

  h1 {
    font-size: 1.5rem;
    line-height: 2.75rem;
  }

  .attribution {
    margin-top: 68%;
  }

  button:hover {
    color: #ffffff;
    background-color: rgb(103, 144, 144);
  }

  .img-logo {
    display: flex;
    justify-content: center;
  }

  .absolute-bg {
    background-image: linear-gradient(
        to bottom,
        rgba(32, 136, 136, 0.4),
        rgba(193, 205, 193, 0.8),
        rgb(164, 147, 114)
      ),
      url('../assets/login_background_mobile.png');
  }
}
