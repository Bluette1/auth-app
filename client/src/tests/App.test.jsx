import { it, describe } from "vitest";
import { render, screen } from "@testing-library/react";
import App from "../App.jsx";
import { BrowserRouter } from "react-router-dom";

import configureStore from "../testutils/store";
import { Provider } from "react-redux";

const AppWithStore = () => (
  <Provider store={configureStore()}>
    <BrowserRouter>
      <App />
    </BrowserRouter>
  </Provider>
);

describe("App", () => {
  it("should render the App", () => {
    render(<AppWithStore />);

    const element = screen.getByText(/secure your profile/i);
    expect(element).toBeInTheDocument();
  });
});
