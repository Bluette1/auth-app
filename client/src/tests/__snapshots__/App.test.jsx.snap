// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should render the App 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
        data-testid="app"
      >
        <div>
          <article
            class="img-logo"
          >
            <img
              alt="logo"
              class="logo"
              src="/src/assets/logo.png"
            />
          </article>
           
          <section
            class="logo-area"
          >
            <h1>
              Secure your profile
            </h1>
          </section>
          <main>
            <article
              class=""
            >
              <button
                class="register-btn"
              >
                Register
              </button>
              <button
                class="login-btn"
              >
                Login
              </button>
            </article>
          </main>
          <footer>
            <p
              class="attribution"
            >
              Design by
               
              <a
                href="https://portfolio-ms-app.onrender.com/"
                target="_blank"
              >
                Top Weaver
              </a>
            </p>
          </footer>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
      data-testid="app"
    >
      <div>
        <article
          class="img-logo"
        >
          <img
            alt="logo"
            class="logo"
            src="/src/assets/logo.png"
          />
        </article>
         
        <section
          class="logo-area"
        >
          <h1>
            Secure your profile
          </h1>
        </section>
        <main>
          <article
            class=""
          >
            <button
              class="register-btn"
            >
              Register
            </button>
            <button
              class="login-btn"
            >
              Login
            </button>
          </article>
        </main>
        <footer>
          <p
            class="attribution"
          >
            Design by
             
            <a
              href="https://portfolio-ms-app.onrender.com/"
              target="_blank"
            >
              Top Weaver
            </a>
          </p>
        </footer>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
